src/encoding/schema/binarystring.ts

```diff
@@ -0,0 +1,59 @@
+import { RawBinaryString } from 'algorand-msgpack';
+import {
+  Schema,
+  MsgpackEncodingData,
+  MsgpackRawStringProvider,
+  JSONEncodingData,
+} from '../encoding.js';
+import { coerceToBytes, bytesToString } from '../binarydata.js';
+
+/* eslint-disable class-methods-use-this */
+
+/**
+ * SpecialCaseBinaryStringSchema is a schema for byte arrays which are encoded
+ * as strings in msgpack and JSON.
+ *
+ * This schema allows lossless conversion between the in memory representation
+ * and the msgpack encoded representation, but NOT between the in memory and
+ * JSON encoded representations if the byte array contains invalid UTF-8
+ * sequences.
+ */
+export class SpecialCaseBinaryStringSchema extends Schema {
```

Comment:
In a msgpack encoded block, some fields from the eval delta are encoded as msgpack strings when they actually contain arbitrary bytes. The affected fields are eval delta logs and global and local state delta keys and values. I have updated the `EvalDelta` and `ValueDelta` classes from `src/types/block.ts` to use this new schema and the new `SpecialCaseBinaryStringMapSchema`.

I go into more detail about the underlying problem in this issue: https://github.com/algorand/go-algorand/issues/5807#issuecomment-1781599209

Unless we introduce more fields into blocks which suffer from this same drawback (which I don't think we want to and we've taken steps to prevent), there should be no further need for these schemas in new fields/classes.